controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -1661,3 +1762,214 @@ int cli_bml::steering_client_measure(uint32_t steeringGroupIndex, const std::str
+            return -1;
+        }
+
+        unsigned int channel_pool_arr[BML_DCS_MAX_CHANNEL_POOL_SIZE] = {0};
+
+        for (size_t i = 0; i < channels_size; i++) {
+            channel_pool_arr[i] = beerocks::string_utils::stoi(channels[i]);
+        }
+
+        ret = bml_set_dcs_continuous_scan_params(ctx, radio_mac.c_str(), dwell_time, interval_time,
+                                                 channel_pool_arr, channel_pool_size);
+    }
+    printBmlReturnVals("bml_set_dcs_continuous_scan_params", ret);
+
+    return 0;
+}
+
+int cli_bml::get_dcs_continuous_scan_params(const std::string &radio_mac)
+{
+    std::cout << "get_dcs_continuous_scan_params" << std::endl;
+
+    int dwell_time = -1, interval_time = -1, channel_pool_size = BML_DCS_MAX_CHANNEL_POOL_SIZE;
+    unsigned int channel_pool[BML_DCS_MAX_CHANNEL_POOL_SIZE] = {0};
+
+    int ret = bml_get_dcs_continuous_scan_params(ctx, radio_mac.c_str(), &dwell_time,
+                                                 &interval_time, channel_pool, &channel_pool_size);
+
+    if (ret == BML_RET_OK) {
+        std::cout << "dcs-continuous-scan-params for mac:" << std::endl
+                  << "dwell_time=" << dwell_time << std::endl
+                  << "interval_time=" << interval_time << std::endl;
+
+        std::cout << "channel_pool=";
+        for (int i = 0; i < channel_pool_size && i < BML_DCS_MAX_CHANNEL_POOL_SIZE; i++) {
+            if (channel_pool[i] > 0) {
+                if (i > 0) {
+                    std::cout << ",";
+                }
+                std::cout << channel_pool[i];
+            }
+        }
+        std::cout << std::endl;
+        std::cout << "channel_pool_size=" << channel_pool_size << std::endl;
+    }
+
+    printBmlReturnVals("bml_get_dcs_continuous_scan_params", ret);
+
+    return 0;
+}
+
+int cli_bml::start_dcs_single_scan(const std::string &radio_mac, int dwell_time,
+                                   int channel_pool_size, const std::string &channel_pool)
+{
+    std::cout << "start_dcs_single_scan" << std::endl;
+
+    if (dwell_time <= 0) {
+        std::cout << __func__ << "invalid input: dwell_time(" << dwell_time << ") <= 0"
+                  << std::endl;
+        return -1;
+    }
+
+    if (channel_pool.length() == 0 || channel_pool_size == BML_DCS_INVALID_PARAM) {
+        std::cout << __func__
+                  << "invalid channel_pool input: channel_pool.length()==" << channel_pool.length()
+                  << ", channel_pool_size==" << channel_pool_size << std::endl;
+        return -1;
+    }
+
+    auto channels      = string_utils::str_split(channel_pool, ',');
+    auto channels_size = channels.size();
+
+    if (channels_size > BML_DCS_MAX_CHANNEL_POOL_SIZE) {
+        std::cout << "size of channel_pool is too big. size=" << channels_size << std::endl;
+        return -1;
+    }
+
+    if (channels_size > 0 && channels_size != (size_t)channel_pool_size) {
```

Comment:
IMO, parentheses in conditionals, even if they are not strictly required, help writing code that is clear and correct

Reply:
this is not maintained across the code - but if you find it more readable, I don't mind adding it

fixed.

Reply:
This was not something to be addressed in this PR but to be discussed if code writing rules are ever revisited.